x-pack/plugins/security_solution/server/lib/telemetry/tasks/detection_rule.ts

```diff
@@ -88,9 +88,10 @@ export function createTelemetryDetectionRuleListsTaskConfig(maxTelemetryBatch: n
         LIST_DETECTION_RULE_EXCEPTION
       );
 
-      batchTelemetryRecords(detectionRuleExceptionsJson, maxTelemetryBatch).forEach((batch) => {
-        sender.sendOnDemand(TELEMETRY_CHANNEL_LISTS, batch);
-      });
+      const batches = batchTelemetryRecords(detectionRuleExceptionsJson, maxTelemetryBatch);
+      for (const batch of batches) {
+        await sender.sendOnDemand(TELEMETRY_CHANNEL_LISTS, batch);
+      }
```

Comment:
NOTE: I changed this from a dangling promise as if we have dangling promises, then promise rejections will happen on the next "tick" of the NodeJS process which could mask errors. Also, this fixes it to make the telemetry preview endpoint deterministic when using a `await` and calling the tasks which is what we want. Otherwise we have to do messy spins waiting for potential data at a latter point.

Reply:
Good to know